heartbeat/nagios4

```diff
@@ -0,0 +1,572 @@
+nagios_validate_all(){
+    ocf_log debug "nagios_validate_all: checking all the config files"
+    # Check that nagios binary exists.
+    if [ ! -f ${OCF_RESKEY_binary} ]; then
+        ocf_log debug "nagios_validate_all: Executable file ${OCF_RESKEY_binary} not found.  Exiting."
+        exit ${OCF_ERR_GENERIC}
+    fi
+    
+    # Check that configuration file exists.
+    if [ ! -f ${OCF_RESKEY_config} ]; then
+         ocf_log debug "nagios_validate_all: Configuration file ${OCF_RESKEY_config} not found.  Exiting."
+        exit ${OCF_ERR_GENERIC}
+    fi
+    
+    # check for plugin installation
+    if [ ! -d ${OCF_RESKEY_plugindir} ]; then
+    	 ocf_log debug "nagios_validate_all: Plugins do not seem to be installed"
+    	 ocf_log debug "nagios_validate_all: installation location should be: ${OCF_RESKEY_plugindir}"
+    	exit ${OCF_ERR_GENERIC}
+    fi
+    # run nagios config check
+	 ocf_log debug "nagios_validate_all: running ${OCF_RESKEY_binary} -v ${OCF_RESKEY_config}"
+    ${OCF_RESKEY_binary} -v ${OCF_RESKEY_config} > /dev/null 2>&1;
+	if [ $? != "0" ]; then
+   	 ocf_log debug "nagios_validate_all: running ${OCF_RESKEY_binary} -v ${OCF_RESKEY_config}"
+    	   return ${OCF_ERR_GENERIC}
+	fi
+}
+
+nagios_monitor(){
+    local rc
+    # exit immediately if configuration is not valid
+    # nagios_validate_all || exit $?
+    # test if nagios is running.
+    get_nagios_status
+    case "$?" in
+        0)
+            rc=$OCF_SUCCESS
+            ocf_log debug "monitor: Resource is running"
+            ;;
+        1)
+            rc=$OCF_NOT_RUNNING
+            ocf_log debug "monitor: Resource is not running"
+            ;;
+        *)
+            ocf_log err "monitor: Resource has failed"
+            exit $OCF_ERR_GENERIC
+    esac
+    return $rc
+}
+
+# WHAT: return TRUE if a process with given PID is running
+# NOTE: input process id
+#       output true or false 
+ProcessRunning() {
+	ocf_log debug  "start: calling ProcessRunning "
+	RunningPID=$1
+	# Use /proc if it looks like it's here...
+	if [ -d /proc -a -d /proc/1 ]; then
+		[ -d /proc/$RunningPID ]
+	else
+		# This assumes we're running as root...
+		kill -s 0 "$RunningPID" >/dev/null 2>&1
+	fi
+}
+
+silent_status() {
+	ocf_log debug  "start: calling silent_status "
+	local OCF_RESKEY_PidFile="/var/nagios/nagios.pid"
+	if [ -f $OCF_RESKEY_PidFile ]; then
+	    local PID=`cat $OCF_RESKEY_PidFile`
+	ocf_log debug  "start: calling silent_status in if pid=$PID  "
+            ProcessRunning `cat $OCF_RESKEY_PidFile`
+	else
+	ocf_log debug  "start: calling silent_status in ELSE  pid=$PID  "
+		: No pid file
+		false
+	fi
+}
+
+
+# WHAT: return running PIDs if any or return false
+# WHY:  
+# NOTE: return 1 or running PIDs numbers
+nagios_running_pid(){
+        RunningPID=`ps auxww | grep nagios | grep -v grep | grep ${OCF_RESKEY_config} | awk '{print $2}'`
+	if [ ${RunningPID}=""  ]; then
```

Comment:
missing spaces before/after `=`, should be `if [ ${RunningPID} = ""  ]`,
or even better `if [ -z ${RunningPID} ]`